### DISCUSSION TITLE
Custom virtual field while resolving proccess

### DISCUSSION THREAD

**Discussion by scion4581** [overblog/GraphQLBundle#828] [Q&A] [answered]
Hello!
Imagine that in resolver we get some entity with help of injected repository
Is there a possibility to map graphql field with some say virtual property or get method from entity?
For example, I want to concatenate two fields and return like graphql field
Some mappers from entity properties/getters to graphql fields
Maybe something similar already exists and I missed that
Thanks

**Comment by Vincz** [comment, marked as answer]
Hi. You have to use annotations for that. It's the only way to have a mapping between PHP objects and GraphQL types. Look @GQL\Field annotation for example.

**Reply by scion4581** [reply]
It's a pity... Together doctrine, validation, Gedmo, serialization, and now @GQL creates an annotation hell
Anyway, thanks for the answer

**Comment by murtukov** [comment]
@scion4581 No, unfortunately there is no such feature out of the box, but it's on our roadmap.
